Add a tool to create a CSV file from safer C++ expectations

rniwa opened the pull request:
#### 6a7f7f6777890036cdec949a9586ec0cdc5aeb55
<pre>
Add a tool to create a CSV file from safer C++ expectations
<a href="https://bugs.webkit.org/show_bug.cgi?id=293250">https://bugs.webkit.org/show_bug.cgi?id=293250</a>

Reviewed by Brianna Fan.

Add a new script to create a CSV file by aggregating safer C++ expectation files.

This PR also adds webkitpy.safer_cpp.checkers to share more code across safer C++ scripts.

* Tools/Scripts/create-csv-from-safer-cpp-expectations: Added.
(parser): Added. Constructs argparser.
(load_expectations): Added. Loads a single expectation file.
(main): Added.
* Tools/Scripts/webkitpy/safer_cpp/__init__.py: Added.
* Tools/Scripts/webkitpy/safer_cpp/checkers.py: Added.
(Checker): Added.
(Checker.__init__): Added.
(Checker.name): Added.
(Checker.description): Added.
(Checker.expectations_path): Added. Returns the path to given project&apos;s expectation file.
(Checker.enumerate): Added. Lists all the checkers in the order.
(Checker.projects): Added. Lists all the WebKit projects with expectation files.
(CHECKERS): Added.

Canonical link: <a href="https://commits.webkit.org/295132@main">https://commits.webkit.org/295132@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f5a53720b6f53d908073b9b4493c93ebc649ab1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104169 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23873 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14193 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/109365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54837 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106209 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24241 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32417 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/109365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107175 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/18831 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/93977 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/109365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/103645 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18643 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/12015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54197 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/88350 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/12073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111751 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/31325 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/23097 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/88143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31689 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/90164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/87803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/32691 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/10444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/25789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16912 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31254 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/31048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/34384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/32608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/Scripts/create-csv-from-safer-cpp-expectations:
```diff
@@ -0,0 +1,74 @@
+        row = [file_name]
+        for checker_type in checker_list:
+            row.append('Has Failures' if checker_type in checker_set else '')
+        print(','.join(row))
```
Do you plan on piping the output of this script to a file or copying and pasting it?
It could be nice to add a `--output-file` arg or similar to create a file automatically.

rniwa replied:
Good point. Will add that option.

briannafan commented on Tools/Scripts/create-csv-from-safer-cpp-expectations:
```diff
@@ -0,0 +1,74 @@
+        '--project', '-p',
+        choices=Checker.projects(),
+        required=True,
+        help='Specify which project expectations you want to update'
```
Will we ever want to generate these files for more than one project at a time?

rniwa replied:
Hm... I think one file per project makes most sense.

briannafan reviewed (approved):
Nice! `Checker` will be great to use in our other scripts :)

rniwa committed:
Add a tool to create a CSV file from safer C++ expectations
https://bugs.webkit.org/show_bug.cgi?id=293250

Reviewed by Brianna Fan.

Add a new script to create a CSV file by aggregating safer C++ expectation files.

This PR also adds webkitpy.safer_cpp.checkers to share more code across safer C++ scripts.

* Tools/Scripts/create-csv-from-safer-cpp-expectations: Added.
(parser): Added. Constructs argparser.
(load_expectations): Added. Loads a single expectation file.
(main): Added.
* Tools/Scripts/webkitpy/safer_cpp/__init__.py: Added.
* Tools/Scripts/webkitpy/safer_cpp/checkers.py: Added.
(Checker): Added.
(Checker.__init__): Added.
(Checker.name): Added.
(Checker.description): Added.
(Checker.expectations_path): Added. Returns the path to given project's expectation file.
(Checker.enumerate): Added. Lists all the checkers in the order.
(Checker.projects): Added. Lists all the WebKit projects with expectation files.
(CHECKERS): Added.

Canonical link: https://commits.webkit.org/295132@main
